src/main/java/org/elasticsearch/client/support/AbstractClient.java

```diff
@@ -548,6 +782,838 @@ public void fieldStats(FieldStatsRequest request, ActionListener<FieldStatsRespo
+        }
+
+        @Override
+        public ActionFuture<RestoreSnapshotResponse> restoreSnapshot(RestoreSnapshotRequest request) {
+            return execute(RestoreSnapshotAction.INSTANCE, request);
+        }
+
+        @Override
+        public void restoreSnapshot(RestoreSnapshotRequest request, ActionListener<RestoreSnapshotResponse> listener) {
+            execute(RestoreSnapshotAction.INSTANCE, request, listener);
+        }
+
+        @Override
+        public RestoreSnapshotRequestBuilder prepareRestoreSnapshot(String repository, String snapshot) {
+            return new RestoreSnapshotRequestBuilder(this, RestoreSnapshotAction.INSTANCE, repository, snapshot);
+        }
+
+
+        @Override
+        public ActionFuture<SnapshotsStatusResponse> snapshotsStatus(SnapshotsStatusRequest request) {
+            return execute(SnapshotsStatusAction.INSTANCE, request);
+        }
+
+        @Override
+        public void snapshotsStatus(SnapshotsStatusRequest request, ActionListener<SnapshotsStatusResponse> listener) {
+            execute(SnapshotsStatusAction.INSTANCE, request, listener);
+        }
+
+        @Override
+        public SnapshotsStatusRequestBuilder prepareSnapshotStatus(String repository) {
+            return new SnapshotsStatusRequestBuilder(this, SnapshotsStatusAction.INSTANCE, repository);
+        }
+
+        @Override
+        public SnapshotsStatusRequestBuilder prepareSnapshotStatus() {
+            return new SnapshotsStatusRequestBuilder(this, SnapshotsStatusAction.INSTANCE);
+        }
+    }
+
+    static class IndicesAdmin implements IndicesAdminClient {
+
+        private final ElasticsearchClient client;
+
+        public IndicesAdmin(ElasticsearchClient client) {
+            this.client = client;
+        }
+
+        @Override
+        public <Request extends ActionRequest, Response extends ActionResponse, RequestBuilder extends ActionRequestBuilder<Request, Response, RequestBuilder>> ActionFuture<Response> execute(Action<Request, Response, RequestBuilder> action, Request request) {
+            return client.execute(action, request);
+        }
+
+        @Override
+        public <Request extends ActionRequest, Response extends ActionResponse, RequestBuilder extends ActionRequestBuilder<Request, Response, RequestBuilder>> void execute(Action<Request, Response, RequestBuilder> action, Request request, ActionListener<Response> listener) {
+            client.execute(action, request, listener);
+        }
+
+        @Override
+        public <Request extends ActionRequest, Response extends ActionResponse, RequestBuilder extends ActionRequestBuilder<Request, Response, RequestBuilder>> RequestBuilder prepareExecute(Action<Request, Response, RequestBuilder> action) {
+            return client.prepareExecute(action);
+        }
+
+        @Override
+        public ThreadPool threadPool() {
+            return client.threadPool();
+        }
```

Comment:
maybe the 4 methods above can be shared, given that they seem repeated across the different clients here?